tiledb/sm/query/query.cc

```diff
@@ -318,6 +331,19 @@ Status Query::set_subarray(const void* subarray) {
   return Status::Ok();
 }
 
+Status Query::submit() {  // Do nothing if the query is completed or failed
+  RETURN_NOT_OK(init());
+  return storage_manager_->query_submit(this);
+}
+
+Status Query::submit_async(
+    std::function<void(void*)> callback, void* callback_data) {
+  callback_ = callback;
```

Comment:
if there is an error on `init()` here should remove the callback / callback assignments, otherwise the object is in a potentially invalid state